libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3844,13 +3846,23 @@ void TypeChecker::endVisit(Literal const& _literal)
 				)
 					// TODO: Is this triggered when the argument is out of range? Test.
 					parameterTypeMessage = "The type of the literal cannot be converted to the parameters of the suffix function.";
+
+				if (
+					auto expParam = dynamic_cast<IntegerType const*>(suffixFunctionType->parameterTypes().at(1));
+					expParam && expParam->isSigned()
+				)
+					exponentTypeMessage = "The type of the exponent must be unsigned for decimal literal suffix functions.";
```

Comment:
Maybe we could also explain why it's disallowed:
```suggestion
					exponentTypeMessage = "The exponent parameter of a literal suffix function must be unsigned. Exponent is always either zero or a negative power of 10 but the parameter represents its absolute value.";
```

Reply:
Done.